homeassistant/components/permobil/sensor.py

```diff
@@ -0,0 +1,224 @@
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=BATTERY_MAX_DISTANCE_LEFT,
+        key="max_distance_left",
+        name="Permobil Full Charge Distance",
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=USAGE_DISTANCE,
+        key="usage_distance",
+        name="Permobil Distance Traveled",
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        device_class=SensorDeviceClass.DISTANCE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=USAGE_ADJUSTMENTS,
+        key="usage_adjustments",
+        name="Permobil Number of Adjustments",
+        native_unit_of_measurement="adjustments",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    PermobilSensorEntityDescription(
+        value_fn=lambda x: x,
+        item=RECORDS_SEATING,
+        key="record_adjustments",
+        name="Permobil Record Number of Adjustments",
+        native_unit_of_measurement="adjustments",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create sensors from a config entry created in the integrations UI."""
+
+    # create the API object from the config
+    coordinator = hass.data[DOMAIN][COORDINATOR][config_entry.entry_id]
+
+    entities = [
+        PermobilSensor(coordinator=coordinator, description=description)
+        for description in SENSOR_DESCRIPTIONS
+    ]
+
+    async_add_entities(entities, update_before_add=True)
+
+
+class PermobilSensor(CoordinatorEntity[MyPermobilCoordinator], SensorEntity):
+    """Representation of a Sensor.
+
+    This implements the common functions of all sensors.
+    """
+
+    _attr_suggested_display_precision: int = 0
+
+    def __init__(
+        self,
+        coordinator: MyPermobilCoordinator,
+        description: PermobilSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator=coordinator)
+        self._coordinator = coordinator
+        self.entity_description = description
+        self._attr_name = str(description.name)
+        self._item = description.item
+        self._value_fn = description.value_fn
+        self._attr_unique_id = f"{coordinator.p_api.email}_{self._item}"
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the value of the sensor."""
+        endpoint = ENDPOINT_LOOKUP[self._item]
+        data = self._coordinator.data.get(endpoint, {}).get(self._item)
```

Comment:
Instead of doing `get()` calls and such, you could pass the data to the lambda you've defined and let the description get the right value.

This will become something like this here:

```suggestion
        return self.entity_description.value_fn(self.coordinator.data, self.entity_description.endpoint)
```

Notice: Adde the endpoint as part of the entity description, removing the need to lookup.